### ISSUE TITLE
Continue orthofinder from making species tree?

### ISSUE THREAD

**Issue by setarehi** [davidemms/OrthoFinder#176]
Dear David,

Sorry to bother you again, but I was almost done but somehow or the other our network here crashed in the middle of a succesfull orthofinder run. As I have to run everything on my own desktop I would really like to continue the orthofinder algorithm where it has ended. Is that possible? It ended in the middle of making the gene trees. I hope that you can help me with this.

Here is my output:
setarehs-mini:orthofinder setareh$ python orthofinder.py 

OrthoFinder version 2.2.1 Copyright (C) 2014 David Emms

SIMPLE USAGE:
Run full OrthoFinder analysis on FASTA format proteomes in <dir>
  orthofinder [options] -f <dir>

Add new species in <dir1> to previous run in <dir2> and run new analysis
  orthofinder [options] -f <dir1> -b <dir2>

OPTIONS:
 -t <int>          Number of parallel sequence search threads [Default = 16]
 -a <int>          Number of parallel analysis threads [Default = 1]
 -M <txt>          Method for gene tree inference. Options 'dendroblast' & 'msa'
                   [Default = dendroblast]
 -S <txt>          Sequence search program [Default = blast]
                   Options: blast, blast_gz, diamond
 -A <txt>          MSA program, requires '-M msa' [Default = mafft]
                   Options: mafft, muscle, mafft
 -T <txt>          Tree inference method, requires '-M msa' [Default = fasttree]
                   Options: mafft, iqtree, fasttree, raxml
 -s <file>         User-specified rooted species tree
 -I <int>          MCL inflation parameter [Default = 1.5]
 -x <file>         Info for outputting results in OrthoXML format
 -p <dir>          Write the temporary pickle files to <dir>
 -1                Only perform one-way sequence search 
 -n <txt>          Name to append to the results directory
 -h                Print this help text

WORKFLOW STOPPING OPTIONS:
 -op               Stop after preparing input files for BLAST
 -og               Stop after inferring orthogroups
 -os               Stop after writing sequence files for orthogroups
                   (requires '-M msa')
 -oa               Stop after inferring alignments for orthogroups
                   (requires '-M msa')
 -ot               Stop after inferring gene trees for orthogroups 

WORKFLOW RESTART COMMANDS:
 -b  <dir>         Start OrthoFinder from pre-computed BLAST results in <dir>
 -fg <dir>         Start OrthoFinder from pre-computed orthogroups in <dir>
 -ft <dir>         Start OrthoFinder from pre-computed gene trees in <dir>

LICENSE:
 Distributed under the GNU General Public License (GPLv3). See License.md

CITATION:
 When publishing work that uses OrthoFinder please cite:
 Emms D.M. & Kelly S. (2015), Genome Biology 16:157

 If you use the species tree in your work then please also cite:
 Emms D.M. & Kelly S. (2017), MBE 34(12): 3267-3278
 Emms D.M. & Kelly S. (2018), bioRxiv https://doi.org/10.1101/267914
setarehs-mini:orthofinder setareh$ python orthofinder.py -M msa -T iqtree -t 4 -b /Volumes/UNTITLED/OrthFinderPrograms/OrthoFinder-2.2.1/Sequences/Results_May02_1/WorkingDirectory 

OrthoFinder version 2.2.1 Copyright (C) 2014 David Emms

4 thread(s) for highly parallel tasks (BLAST searches etc.)
1 thread(s) for OrthoFinder algorithm

Checking required programs are installed
----------------------------------------
Test can run "mcl -h" - ok
Test can run "mafft /Volumes/UNTITLED/OrthFinderPrograms/OrthoFinder-2.2.1/Sequences/Results_May02_1/WorkingDirectory//_dependencies_check/SimpleTest.fa" - ok
Test can run "iqtree" - ok
Using previously calculated BLAST results in /Volumes/UNTITLED/OrthFinderPrograms/OrthoFinder-2.2.1/Sequences/Results_May02_1/WorkingDirectory/

Running OrthoFinder algorithm
-----------------------------
2018-05-30 08:41:42 : Initial processing of each species
2018-05-30 14:58:55 : Initial processing of species 0 complete
2018-05-30 17:50:36 : Initial processing of species 1 complete
2018-05-30 22:17:59 : Initial processing of species 2 complete
2018-05-31 02:27:15 : Initial processing of species 3 complete
2018-05-31 08:12:55 : Initial processing of species 4 complete
2018-05-31 10:51:01 : Initial processing of species 5 complete
2018-05-31 13:31:15 : Initial processing of species 6 complete
2018-05-31 16:20:01 : Initial processing of species 7 complete
2018-05-31 18:58:11 : Initial processing of species 8 complete
2018-05-31 22:03:56 : Initial processing of species 9 complete
2018-06-01 14:41:59 : Initial processing of species 10 complete
2018-06-01 22:56:44 : Initial processing of species 11 complete
2018-06-05 06:38:32 : Initial processing of species 12 complete
2018-06-05 12:56:39 : Initial processing of species 13 complete
2018-06-05 18:26:45 : Initial processing of species 14 complete
2018-06-05 21:33:47 : Initial processing of species 15 complete
2018-06-06 00:20:22 : Initial processing of species 16 complete
2018-06-06 03:30:03 : Initial processing of species 17 complete
2018-06-06 08:29:04 : Initial processing of species 18 complete
2018-06-06 15:55:52 : Initial processing of species 19 complete
2018-06-06 21:24:16 : Initial processing of species 20 complete
2018-06-06 23:34:21 : Initial processing of species 21 complete
2018-06-07 02:14:59 : Initial processing of species 22 complete
2018-06-07 04:49:13 : Initial processing of species 23 complete
2018-06-07 15:34:15 : Initial processing of species 24 complete
2018-06-07 21:18:51 : Initial processing of species 25 complete
2018-06-08 01:13:25 : Initial processing of species 26 complete
2018-06-08 03:39:26 : Initial processing of species 27 complete
2018-06-08 06:03:50 : Initial processing of species 28 complete
2018-06-08 08:44:43 : Initial processing of species 29 complete
2018-06-08 11:46:16 : Initial processing of species 30 complete
2018-06-08 15:20:27 : Initial processing of species 31 complete
2018-06-08 18:53:19 : Initial processing of species 32 complete
2018-06-08 22:55:34 : Initial processing of species 33 complete
2018-06-09 01:13:08 : Initial processing of species 34 complete
2018-06-09 03:40:49 : Initial processing of species 35 complete
2018-06-09 06:09:13 : Initial processing of species 36 complete
2018-06-09 08:39:08 : Initial processing of species 37 complete
2018-06-09 10:56:57 : Initial processing of species 38 complete
2018-06-09 13:06:09 : Initial processing of species 39 complete
2018-06-09 20:01:41 : Connected putatitive homologs
2018-06-09 20:13:04 : Written final scores for species 0 to graph file
2018-06-09 20:16:30 : Written final scores for species 1 to graph file
2018-06-09 20:25:52 : Written final scores for species 2 to graph file
2018-06-09 20:35:32 : Written final scores for species 3 to graph file
2018-06-09 20:51:30 : Written final scores for species 4 to graph file
2018-06-09 20:55:20 : Written final scores for species 5 to graph file
2018-06-09 20:57:43 : Written final scores for species 6 to graph file
2018-06-09 21:00:21 : Written final scores for species 7 to graph file
2018-06-09 21:02:58 : Written final scores for species 8 to graph file
2018-06-09 21:06:07 : Written final scores for species 9 to graph file
2018-06-10 01:47:13 : Written final scores for species 10 to graph file
2018-06-10 02:49:27 : Written final scores for species 11 to graph file
2018-06-12 07:44:13 : Written final scores for species 12 to graph file
2018-06-12 15:31:47 : Written final scores for species 13 to graph file
2018-06-12 15:45:13 : Written final scores for species 14 to graph file
2018-06-12 15:49:12 : Written final scores for species 15 to graph file
2018-06-12 15:51:46 : Written final scores for species 16 to graph file
2018-06-12 15:54:37 : Written final scores for species 17 to graph file
2018-06-12 16:04:13 : Written final scores for species 18 to graph file
2018-06-12 16:21:35 : Written final scores for species 19 to graph file
2018-06-12 16:34:10 : Written final scores for species 20 to graph file
2018-06-12 16:36:46 : Written final scores for species 21 to graph file
2018-06-12 16:39:18 : Written final scores for species 22 to graph file
2018-06-12 16:41:35 : Written final scores for species 23 to graph file
2018-06-12 17:28:29 : Written final scores for species 24 to graph file
2018-06-12 17:56:35 : Written final scores for species 25 to graph file
2018-06-12 18:03:54 : Written final scores for species 26 to graph file
2018-06-12 18:06:13 : Written final scores for species 27 to graph file
2018-06-12 18:08:30 : Written final scores for species 28 to graph file
2018-06-12 18:10:46 : Written final scores for species 29 to graph file
2018-06-12 18:13:00 : Written final scores for species 30 to graph file
2018-06-12 18:16:46 : Written final scores for species 31 to graph file
2018-06-12 18:22:49 : Written final scores for species 32 to graph file
2018-06-12 18:31:59 : Written final scores for species 33 to graph file
2018-06-12 18:34:50 : Written final scores for species 34 to graph file
2018-06-12 18:37:15 : Written final scores for species 35 to graph file
2018-06-12 18:39:48 : Written final scores for species 36 to graph file
2018-06-12 18:42:20 : Written final scores for species 37 to graph file
2018-06-12 18:44:38 : Written final scores for species 38 to graph file
2018-06-12 18:46:44 : Written final scores for species 39 to graph file
2018-06-12 23:45:43 : Ran MCL

Writing orthogroups to file
---------------------------
A duplicate accession was found using just first part: Alp215600
Tried to use only the first part of the accession in order to list the sequences in each orthogroup
more concisely but these were not unique. The full accession line will be used instead.

Orthogroups have been written to tab-delimited files:
   /Volumes/UNTITLED/OrthFinderPrograms/OrthoFinder-2.2.1/Sequences/Results_May02_1/WorkingDirectory/Orthogroups_3.csv
   /Volumes/UNTITLED/OrthFinderPrograms/OrthoFinder-2.2.1/Sequences/Results_May02_1/WorkingDirectory/Orthogroups_3.txt (OrthoMCL format)
   /Volumes/UNTITLED/OrthFinderPrograms/OrthoFinder-2.2.1/Sequences/Results_May02_1/WorkingDirectory/Orthogroups_3_UnassignedGenes.csv
2018-06-12 23:46:56 : Done orthogroups

Analysing Orthogroups
=====================
A duplicate accession was found using just first part: Alp215600
Tried to use only the first part of the accession in order to list the sequences in each orthogroup
more concisely but these were not unique. The full accession line will be used instead.

Species tree: Using 282 orthogroups with minimum of 22.5% of species having single-copy genes in any orthogroup

Inferring multiple sequence alignments for species tree
-------------------------------------------------------
2018-06-13 00:53:28 : Done 0 of 282
2018-06-13 03:21:31 : Done 100 of 282
2018-06-13 04:17:06 : Done 200 of 282

Inferring remaining multiple sequence alignments and gene trees
---------------------------------------------------------------
2018-06-13 04:37:37 : Done 0 of 32601

**Comment by davidemms** [comment]
Hi

You can start from the calculated orthogroups using '-fg':
`python orthofinder.py -fg /Volumes/UNTITLED/OrthFinderPrograms/OrthoFinder-2.2.1/Sequences/Results_May02_1/WorkingDirectory/CLUSTERS_PAIRS_FILENAME -t 4`

Where CLUSTERS_PAIRS_FILENAME is the file `clusters_OrthoFinder_..._id_pairs.txt` file from the run you want to restart, i.e. from around 2018-06-12 23:46:56.

All the best
David

**Comment by davidemms** [comment]
It uses the a method called STAG (Species Tree from All Genes) that allows the inference of the species tree even without sets of one-to-one orthologues: https://www.biorxiv.org/content/early/2018/02/19/267914. Lack of complete sets of one-to-one orthologues can frequently occur because of gene duplication and loss across the clade of species.

**Comment by davidemms** [comment]
See the response in this thread: https://github.com/davidemms/OrthoFinder/issues/186

Apologies for the delay in replying, I've been away from work. I'll close this thread and we can use the other thread for resolving this issue.

All the best
David

**Comment by KristinaGagalova** [comment]
Hi, 
Related to this question but not exactly the same.
I have a set of species that I want to group together and a sample which is an introgress.
I am aware that this last sample can create some noise in the analysis so I have maually remapped the introgress sequences to the species that I have grouped together with Orthofinder. I have created a core set of OGs with pure species and assigned the introgress to those manually.
Is there a possible way to start the analysis from that stage, after I have the new orthogroups? I have prepared separate fasta file for each OG and I would like Orthofinder to continue from that. I don't have a clustering file from mcl as for the -fg option
Thank you in advance
